server/server.go

```diff
@@ -519,6 +519,12 @@ func killConn(conn *clientConn, query bool) {
 	}
 
 	conn.mu.RLock()
+	for _, resultSet := range conn.mu.resultSets {
+		// resultSet.Close() is reentrant so it's safe to kill a same connID multiple times
+		if err := resultSet.Close(); err != nil {
+			logutil.Logger(context.Background()).Error("close result set error", zap.Uint32("connID", conn.connectionID), zap.Error(err))
```

Comment:
could we move `Close` operation out of hold lock area? just get resultSets ref in lock scope

Reply:
updated